src/optical_spi.c

```diff
@@ -33,10 +33,16 @@ bit 4-0 = well number (0-31)
 
 #include "optical_spi.h"
 
-// volatile needed?
+// not sure if `optical_spi.c` should be creating the CAN messages
+// CAN messages to transmit
+queue_t tx_msg_queue;
+
+// don't need to handle CAN messages, since the main loop should handle them
+// -> use optical_spi.c to collect sensor readings and put them in a new CAN message
 
 // tracking if SPI in progress
 bool spi_in_progress = false;
+uint8_t current_cmd = 0;
```

Comment:
You don't need this variable to keep track of current command, because you are only doing the asynchronous checking for get reading commands. The `spi_in_progress` and `current_well_info` variables only apply to get reading commands, because for other commands you will just block to wait for the response because they will execute very quickly.